x/auth/signing/cache.go

```diff
@@ -0,0 +1,82 @@
+package signing
+
+import (
+	lru "github.com/hashicorp/golang-lru/v2"
+)
+
+var signatureCache *Cache
+
+const (
+	TxHashLen        = 32
+	AddressStringLen = 2 + 20*2
+)
+
+func init() {
+	// used for ut
+
+	cache, err := lru.New[string, []byte](500)
+	if err != nil {
+		panic(err)
+	}
+	defaultCache := &Cache{
+		data: cache,
+	}
+	signatureCache = defaultCache
+}
+
+// InitSignatureCache initializes the signature cache.
+// signature verification is one of the most expensive parts in verification
+// by caching it we avoid needing to verify the same signature multiple times
+func NewSignatureCache() {
+	// 500 * (32 + 42) = 37.5KB
+	cache, err := lru.New[string, []byte](500)
+	if err != nil {
+		panic(err)
+	}
+	signatureCache = &Cache{
+		data: cache,
+	}
+}
```

Comment:
This code looks out of place. The signature says `NewSignatureCache` so I'd expect it to return a cache, but also the comment says `InitSignatureCache` then sets the global value, we have a couple of options for this:
1. InitSignatureCache
```go
var initSigCacheOnce sync.Once

func InitSignatureCache() {
      initSigCacheOnce.Do(func() {
         cache, err := lru.New[string, []byte](500)
         if err != nil {
             panic(err)
         }
         signatureCache = &Cache{data: cache}
     })
}
```

and then use it in the `func init()` so
```go
func init() {
    InitSignatureCache()
}
```

Reply:
Am gonna send the follow-up in some updates in a commit shortly.

Reply:
thank you, made the changes!!